jwst/resample/resample_utils.py

```diff
@@ -124,17 +124,11 @@ def shape_from_bounding_box(bounding_box):
 def calc_gwcs_pixmap(in_wcs, out_wcs, shape=None):
     """ Return a pixel grid map from input frame to output frame.
     """
-    if shape:
-        bb = wcs_bbox_from_shape(shape)
-        log.debug("Bounding box from data shape: {}".format(bb))
-    else:
-        bb = in_wcs.bounding_box
-        log.debug("Bounding box from WCS: {}".format(in_wcs.bounding_box))
-
-    grid = gwcs.wcstools.grid_from_bounding_box(bb)
-    pixmap = np.dstack(reproject(in_wcs, out_wcs)(grid[0], grid[1]))
+    if shape is not None and not np.array_equiv(shape, in_wcs.array_shape):
+        in_wcs = deepcopy(in_wcs)
```

Comment:
Is a `deepcopy` necessary here?

Reply:
I am not sure. It felt to be a safer way because this method sets `array_shape` property of the WCS object. If there is no drawback in modifying inputs, `deepcopy` can be omitted.

Reply:
What happens here if `shape is None` ?